### ISSUE TITLE
PrometheusPushGateway cuts off current valid metrics  

### ISSUE THREAD

**Issue by vkorchik** [kamon-io/Kamon#912]
Hi,
I am using Kamon version 2.1.9

I have found it very strange but `PrometheusPushGatewayReporter` is created with a hardcoded stale period of 365 days, and this forces all my current metrics to be cleaned [here](https://github.com/kamon-io/Kamon/blob/8ad1433ab97e0121d4b35cffa43822e9fef832ea/core/kamon-core/src/main/scala/kamon/metric/PeriodSnapshot.scala#L141). Here we have from debug:
```
stalePeriod = PT8760H // which is 365 days
currentInstant = 2024-12-18T00:00:00Z
cutoff = 2023-12-19T00:00:00Z
```

So here i will give you my config file, and the way i emit metrics. 
```
// application.conf

kamon {
  prometheus {
    start-embedded-http-server = no
    pushgateway.api-url = "http://localhost:9091/metrics/job/example"
  }

  modules {
    prometheus-reporter.enabled = false
    pushgateway-reporter.enabled = true
  }
}


// Main.scala

Kamon.init()
Kamon.counter("example-counter-1").withoutTags().increment(1)
Kamon.counter("example-counter-1").withoutTags().increment(1)
```

PS:
The usual `PrometheusReporter` works well, since stale period there is 5 years as well as period, so cutoff date is basically the now date.

**Comment by muej** [comment]
I faced this issue too and have tracked it down to be an issue in the `PeriodSnapshot.Accumulator` and I guess this [change](https://github.com/kamon-io/Kamon/commit/406c9069d331d56f32c1efcdb25cddbc2e4d4eb5) was not applied here [PrometheusPushgatewayReporter.scala#L34](https://github.com/kamon-io/Kamon/blob/master/reporters/kamon-prometheus/src/main/scala/kamon/prometheus/PrometheusPushgatewayReporter.scala#L34).

**Comment by muej** [comment]
If it is of any help, my configuration can be found on [gitter](https://gitter.im/kamon-io/Kamon?at=602ce8f1a8c6a77848153a25)

**Comment by muej** [comment]
#822 seems to be related.

**Comment by vkorchik** [comment]
@muej, yes you are right, the problem is exactly in instantiating the snapshot accumulator.
But PR is there for more than 6 months, hope it will be merged anytime soon.

**Comment by SimunKaracic** [comment]
Fixed in version 2.1.14!
